daemon/mgr/container.go

```diff
@@ -322,13 +328,12 @@ func (mgr *ContainerManager) Create(ctx context.Context, name string, config *ty
 	networkMode := config.HostConfig.NetworkMode
 	if networkMode == "" {
 		config.HostConfig.NetworkMode = "bridge"
-		container.Config.NetworkDisabled = true
 	}
 	container.NetworkSettings = new(types.NetworkSettings)
 	if len(config.NetworkingConfig.EndpointsConfig) > 0 {
 		container.NetworkSettings.Networks = config.NetworkingConfig.EndpointsConfig
 	}
-	if container.NetworkSettings.Networks == nil && networkMode != "" && !IsContainer(networkMode) {
+	if container.NetworkSettings.Networks == nil && !IsContainer(config.HostConfig.NetworkMode) {
```

Comment:
I found that here we have so much blocks about network handling in this Create function. And this make the code much much longer and a little bit difficult to read. Is there any way to encapsulate these blocks into a single function? @shaloulcy 
Maybe in another pull request.